feature/security/gnsi/authz/tests/authz_test.go

```diff
@@ -0,0 +1,752 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authz_test performs functional tests for authz service
+package authz_test
+
+import (
```

Comment:
It doesn't seems like streaming validation is implemented. Do you plan to do it in a follow up PR? 

```
NOTE: regarding gNMI OC validation:
  * Everytime a gRPC call (including gNSI calls themselves) is allowed or denied, the following OC leaves should be validated:
    * `/system/grpc-servers/grpc-server/authz-policy-counters/rpcs/rpc[name]/state/name` is the matched request path, e.g. "/gribi.gRIBI/Get"
    * `/system/grpc-servers/grpc-server/authz-policy-counters/rpcs/rpc/rpc[name]/state/access-accepts` increments if the rpc call is allowed.
    * `/system/grpc-servers/grpc-server/authz-policy-counters/rpcs/rpc/rpc[name]/state/access-rejects` increments if the rpc call is denied.
    * `/system/grpc-servers/grpc-server/authz-policy-counters/rpcs/rpc/rpc[name]/state/last-access-accept` reflects the timestamp of the method call.
    * `/system/grpc-servers/grpc-server/authz-policy-counters/rpcs/rpc/rpc[name]/state/last-access-reject` reflects the timestamp of the method call.
  * Everytime a valid policy is pushed (even it's not finalized), the following OC leaves should be validated:
    * `/system/grpc-servers/grpc-server/state/authz-policy-version` = `UploadRequest.version` in the API proto.
    * `/system/grpc-servers/grpc-server/state/authz-policy-created-on` = `UploadRequest.created_on` (in terms of represented time).
  * Everytime a valid policy is automatically rolled back, the following OC leaves should be validated:
    * `/system/grpc-servers/grpc-server/state/authz-policy-version` = `UploadRequest.version` of the previous request (the one rollback to).
    * `/system/grpc-servers/grpc-server/state/authz-policy-created-on` = `UploadRequest.created_on` of the previous request (the one rollback to).
  * An invalid policy should not trigger the following OC leaf updates:
    * `/system/grpc-servers/grpc-server/state/authz-policy-version`
    * `/system/grpc-servers/grpc-server/state/authz-policy-created-on`
```

Reply:
We will have a followup PR when we have implementation ready for them. I also noticed even ondatra misses support for some of them.

Reply:
Thanks. Please help Mark TODO in both the README and the code.